src/OrbitVulkanLayer/TimerQueryPool.h

```diff
@@ -76,30 +81,24 @@ class TimerQueryPool {
     return device_to_query_pool_.at(device);
   }
 
-  // Tries to find a free query slot in the device's pool. It returns `false` if no slot was found
-  // (all slots are occupied) and true otherwise.
-  // In case it successfully found a slot, the index will be written to the given `allocated_index`.
+  // Returns a free query slot from the device's pool if one still exists. It returns `false` if all
+  // slots are occupied and true otherwise. If successful, the index will be written to the given
+  // `allocated_index`.
   //
   // Note that the pool must be initialized using `InitializeTimerQueryPool` before.
   // See also `ResetQuerySlots` to make occupied slots available again.
   [[nodiscard]] bool NextReadyQuerySlot(VkDevice device, uint32_t* allocated_index) {
     absl::WriterMutexLock lock(&mutex_);
-    CHECK(device_to_potential_next_free_index_.contains(device));
```

Comment:
That should probably be a CHECK(device_to_free_slots_.contains(device));

Reply:
Yes, good idea. Done.

Reply:
Thanks!